### ISSUE TITLE
isOAuthRequest always true

### ISSUE THREAD

**Issue by aesidau** [node-facebook/facebook-node-sdk#38]
I was going through the code, and came across this line in src/fb.js ..
```
isOAuthRequest = /^oauth.*/.test('oauth/');
```
Since this always sets `isOAuthRequest` to true, I think it should be
```
isOAuthRequest = /^oauth.*/.test(path);
```
However, if it's been like this for ages and hasn't caused a problem for anyone, then there may not be a strong case for changing it.

**Comment by dantman** [comment]
You're right. This is the code that uses it:
```js
if ( isOAuthRequest && response && response.statusCode === 200 &&
	response.headers && /.*text\/plain.*/.test(response.headers['content-type'])) {
	cb(parseOAuthApiResponse(body));
```

The overall effect is that text/plain responses (which typically aren't returned by the Graph API) are parsed using the querystring pattern used to parse OAuth responses.

Since it doesn't really do any harm, we might as well wait to fix it till 2.x when I plan to get rid of other things like fql and legacy rest. Plus some other legacy behaviors like `FacebookApiException` and `version` being present on `Facebook` instances (they should be taken from exports). And probably drop support for 0.10/0.12.

**Comment by dantman** [comment]
Actually, another discovery.

This changes in v2.3 of the API:
> [Oauth Access Token] Format - The response format of https://www.facebook.com/v2.3/oauth/access_token returned when you exchange a code for an access_token now return valid JSON instead of being URL encoded. The new format of this response is {"access_token": {TOKEN}, "token_type":{TYPE}, "expires_in":{TIME}}. We made this update to be compliant with section 5.1 of RFC 6749.

So as of `v2.3` the response changes to JSON and that test becomes useless. Sadly v2.2 will still be available till March 25, 2017.

**Comment by dantman** [comment]
This is fixed in `2.0.0-alpha1`.
